packages/authentication/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/authentication/handlers/GoogleAuthProviderHandler.java

```diff
@@ -105,12 +162,50 @@ public void handleAuthorizationResultError(@NonNull Exception exception) {
         lastIdToken = null;
     }
 
+    public void signIn(final PluginCall call) {
+        Boolean useCredentialManagerRaw = call.getBoolean("useCredentialManager");
+        boolean useCredentialManager = (useCredentialManagerRaw != null) ? useCredentialManagerRaw : true;
+
+        if (useCredentialManager) {
+            signInOrLink(call, false);
+        } else {
+            mGoogleSignInClient = buildGoogleSignInClient(call);
+            Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+            pluginImplementation.startActivityForResult(call, signInIntent, "handleGoogleAuthProviderSignInActivityResult");
+        }
+    }
+
     public void link(final PluginCall call) {
-        signInOrLink(call, true);
+        Boolean useCredentialManagerRaw = call.getBoolean("useCredentialManager");
+        boolean useCredentialManager = (useCredentialManagerRaw != null) ? useCredentialManagerRaw : true;
+
+        if (useCredentialManager) {
+            signInOrLink(call, true);
+        } else {
+            mGoogleSignInClient = buildGoogleSignInClient(call);
+            Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+            pluginImplementation.startActivityForResult(call, signInIntent, "handleGoogleAuthProviderLinkActivityResult");
```

Comment:
Let's move this code into the `signInOrLink` method to keep this method "clean".